### ISSUE TITLE
Support syntax highlighting chained tagged templates

### ISSUE THREAD

**Issue by SleepWalker** [gandm/language-babel#425]
Hello,

is it possible to support syntax highlighting for the chained tagged templates?

For example for the case, when you want a function, that will render one or another template depending on desktop/mobile version:

```js
  ${css.version`
    right: -25px;
    top: 15px;
    width: 90px;
    font-size: 13px;
    font-weight: 900;
    height: 16px;
    line-height: 16px;

    &:lang(pl),
    &:lang(it) {
      font-size: 10px;
    }
  ``
    right: -16px;
    top: 13px;
    width: 60px;
    font-size: 9px;
    font-weight: 400;
    line-height: 13px;
    letter-spacing: 1px;
    height: 13px;
  `};
```

**Comment by gandm** [comment]
See https://github.com/gandm/language-babel/issues/419 for solution.

**Comment by SleepWalker** [comment]
That thing covers different case. I need the scope to be propagated to the consequent templates. Here is how it looks like now:

![shutter-04-10-2017-01](https://user-images.githubusercontent.com/2536916/31165566-3f2f685c-a8f4-11e7-9bef-8d737cc2f858.png)

The first template is colored, but the second one - not.

**Comment by gandm** [comment]
Oh! I missed the follow-on template. I would have to change the generated language grammar to account for this variation. The above is interesting syntax; what is the use case for having two templates like this?

**Comment by SleepWalker** [comment]
In project, that I'm working on there is very usual case, that you write some css and then add some tweaks for mobile and desktop `version` (the app is not fully adaptive and there may be huge differences in layouts etc). I've created the following helpers:

```jsx
  ${css.desktop`
    height: 25px;
    font-size: 10px;
    line-height: 25px;

    &::before {
      border-top: 25px solid transparent;
    }
  `};

  ${css.mobile`
    height: 20px;
    font-size: 9px;
    line-height: 20px;
    font-weight: 700;

    &::before {
      border-top: 20px solid transparent;
    }
  `};
```

Because desktop and mobile are frequently used together, I've thought to create a helper for such cases. That is how `css.version` was created.

I've already decided to abandon this idea (at least for the current moment), because `css.desktop` is more verbose, but it looks more clear compared to `css.version` followed by two templates (and there is no guarantee, that I won't get the new version in future, e.g. `smartWatch` :D).

---

But... I think, that this pattern may occur somewhere else and it is logical to expect chained templates to have some syntax highlighting too.

Actually, there may be another use case with different syntax in templates e.g.:

```js
const component = render`
  <div class="my-component">
    Hello world
  </div>
``
.my-component {
  background: red;
  color: #fff;
}
`;
```

(looks like I have invented the brand new template engine for js :D)

I don't know whether this feature is hard to implement (have no experience with grammars and etc) and, as you saw, there may be too much combinations. So I've create this issue more as proposal and discussion.
